### PR TITLE
Inclusion of salvage and updates to existing datasets

### PR DESCRIPTION
**Pull request by trinhxuann**
Salvage is now included.

All datasets now have the functionality to scrape online data, theoretically from sources that hosts the most up to date version of the publicly available data.

### CODE REVIEW

File: data-raw/DJFMP.R [hunk 1]
```diff
@@ -77,23 +126,26 @@ DJFMP <-  bind_rows(
...
+         Count=ifelse(is.infinite(Count), Total, Count))%>%
+  dplyr::filter(Length!=0 | is.na(Length))%>%
   select(-Total, -TotalMeasured, -Group)%>%
-  left_join(DJFMP_stations, by = "Station") %>%
+  left_join(DJFMP_stations, by = c("Station"="StationCode")) %>%
   # Add species names
-  left_join(Species %>%
-              select(USFWS_Code, Taxa) %>%
-              filter(!is.na(USFWS_Code)),
+  left_join(USFWS_Species,
```
Comment (sbashevkin) [inline data-raw/DJFMP.R:134]: why is this used instead of our unified species table?

File: data-raw/DJFMP.R [hunk 2]
```diff
@@ -77,23 +126,26 @@ DJFMP <-  bind_rows(
...
             by=c("OrganismCode"="USFWS_Code")) %>%
   mutate(SampleID=paste(Source, SampleID), # Add variable for unique (across all studies) sampleID
-         Taxa=str_remove(Taxa, " \\((.*)"))%>% # Remove life stage info from Taxa names
+         #Taxa=str_remove(Taxa, " \\((.*)")
```
Comment (sbashevkin) [inline data-raw/DJFMP.R:137]: why was this done?

File: data-raw/SLS.R [hunk 3]
```diff
@@ -167,16 +182,16 @@ SLS <- waterInfo %>%
...
+            by = c("Date", "Station", "Tow", "FishCode"),multiple="all") %>%
   # Adding in taxa name based on Species Code.csv file
   left_join(Species %>%
               select(TMM_Code,
                      Taxa) %>%
-              filter(!is.na(TMM_Code)),
+              dplyr::filter(!is.na(TMM_Code)),
             by = c("FishCode"="TMM_Code")) %>%
-  left_join(SLSTables$`20mm Stations`,
+  left_join(SLSTables$"20mm Stations",
```
Comment (sbashevkin) [inline data-raw/SLS.R:194]: Does this work? I thought you had to use backticks after a $ instead of quotes?

File: data-raw/Salvage.R [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  # 16 instances of NA counts from the CVP because
+  # Removes just the four CatchRowID
+  filter(!(lengthNAs == 1 & (LengthFrequency == 0 | is.na(LengthFrequency))
+           # & Method == "Normal count"
+           )) %>%
+  # Creating a SampleID. Want to retain SampleRowID from database but also distinguish each building at the SWP
+  {
+    left_join(., distinct(.data = ., Source, Facility, Salvage_building, SampleRowID) %>%
+                group_by(Facility, SampleRowID) %>%
+                mutate(SampleID = paste(Source, SampleRowID, 1:n())),
```
Comment (sbashevkin) [inline data-raw/Salvage.R:183]: Do you need an ungroup after this line?

**Review by sbashevkin — CHANGES_REQUESTED** [review]
Thanks so much for doing this and sorry this took me so long. It looks pretty good. I did have some comments and requests for changes. Could you please address each of those?

File: data-raw/20mm.R [hunk 5]
```diff
@@ -188,25 +228,18 @@ TMM<-TMM%>%
...
 								-LengthFrequency)
-nrow(TMM)
-ncol(TMM)
-names(TMM)
-
-
 
 ## Save compressed data to /data:
-usethis::use_data(TMM, TMM_measured_lengths, overwrite=TRUE)
+usethis::use_data(TMM, TMM_measured_lengths, overwrite=TRUE, compress="xz")
```
[the remark this answers is not in the document]
Comment (trinhxuann) [reply data-raw/20mm.R:243]: Went back and added this to all data scripts

File: data-raw/Suisun/SuisunMarshFish2020_1_14_23.zip
[the remark this answers is not in the document]
Comment (trinhxuann) [reply data-raw/Suisun/SuisunMarshFish2020_1_14_23.zip:1]: Removed. Checked for other database files and found none

File: data-raw/salvage/salvageWebsite_1993.RData
[the remark this answers is not in the document]
Comment (trinhxuann) [reply data-raw/salvage/salvageWebsite_1993.RData:1]: Deleted this data file. Checked for others and found none.

Forgot to update the gitignore before pushing. Will do this next, seeing what would be appropriate for that file.

File: data-raw/DJFMP.R (hunk 1 above)
Comment (trinhxuann) [reply data-raw/DJFMP.R:134]: USFWS_Species is just a subset of the Species table apparently. Seems as though it was done to limit the species codes to USFWS only and not introduce lines of NAs when joining, i.e., species codes from other surveys that DJFMP doesn't use.

Left as is for now.

File: data-raw/DJFMP.R (hunk 2 above)
Comment (trinhxuann) [reply data-raw/DJFMP.R]: Added this back in. Did do a comparison with `all.equal()` and it showed no differences.

File: data-raw/SLS.R (hunk 3 above)
Comment (trinhxuann) [reply data-raw/SLS.R]: Apparently it does. I agree it looks really weird though. Regardless, removed this as this table is now replaced with `SLS Stations` in the SLS Access db. Updated the code accordingly

File: data-raw/salvage/salvageWebsite_1993.RData
[the remark this answers is not in the document]
Comment (trinhxuann) [reply data-raw/salvage/salvageWebsite_1993.RData:1]: Updated gitignore now to ignore data files that compares the salvage dataset

**Review by sbashevkin — APPROVED** [review]

**Pull request merged**